src/json-rpc-socket.ts

```diff
@@ -8,7 +8,7 @@ import type { JsonRpcRequest, JsonRpcResponse } from "./lib/json-rpc.js";
 const CONNECT_TIMEOUT = 3_000;
 const RESPONSE_TIMEOUT = 30_000;
 
-export interface JSONRPCSocketOptions {
+export interface JsonRpcSocketOptions {
```

Comment:
Nit: consider renaming to `JsonRpcClientOptions` for consistency. Though, I liked `JsonRpcSocket` because it implies there is a persistent web socket under the hood, else it could be interpreted as an HTTP client looking at just the name.

Reply:
I ended up renaming `JsonRpcClient` to `JsonRpcSocket` so kept this as socket options as well.